OpenRA.Mods.Common/UtilityCommands/Documentation/DocumentationHelpers.cs

```diff
@@ -46,14 +57,96 @@ public static IEnumerable<ExtractedClassFieldInfo> GetClassFieldInfos(Type type,
 				});
 		}
 
-		public static IEnumerable<ExtractedEnumInfo> GetRelatedEnumInfos(HashSet<Type> relatedEnumTypes)
+		public static IEnumerable<ExtractedEnumInfo> GetRelatedEnumInfos(
+			HashSet<Type> relatedEnumTypes, Cache<string, IReadOnlyDictionary<string, ImmutableArray<string>>> pdbTypesCache)
 		{
 			return relatedEnumTypes.OrderBy(t => t.Name).Select(type => new ExtractedEnumInfo
 			{
 				Namespace = type.Namespace,
 				Name = type.Name,
+				Filename = GetSourceFilenameForType(type, pdbTypesCache),
 				Values = Enum.GetNames(type).ToDictionary(x => Convert.ToInt32(Enum.Parse(type, x), NumberFormatInfo.InvariantInfo), y => y)
 			});
 		}
+
+		public static Cache<string, IReadOnlyDictionary<string, ImmutableArray<string>>> CreatePdbTypesCache()
+		{
+			return new Cache<string, IReadOnlyDictionary<string, ImmutableArray<string>>>(BuildTypeMap);
+		}
+
+		public static string GetSourceFilenameForType(Type type,
+			Cache<string, IReadOnlyDictionary<string, ImmutableArray<string>>> pdbTypesCache)
+		{
+			foreach (var file in pdbTypesCache[type.Assembly.Location])
+				foreach (var t in file.Value)
+					if (t.EndsWith($".{type.Name}", StringComparison.InvariantCultureIgnoreCase))
+						return file.Key;
+
+			return "(unknown)";
+		}
+
+		// Source: ChatGPT because nothing else could have guided me through the existential horrors of dealing with PDBs.
```

Comment:
Could we please rewrite this method to align with OpenRA style conventions (variable naming, comment style are the main things)